test/integration/thruster.cc

```diff
@@ -331,13 +332,13 @@ void ThrusterTest::TestWorld(const std::string &_world,
     EXPECT_NEAR(0.0, angVel.Y(), _baseTol);
     EXPECT_NEAR(0.0, angVel.Z(), _baseTol);
   }
-
+  ASSERT_FALSE(modelPoses.empty());
```

Comment:
I feel this is correct as it's checking if the `modelPoses` has some value before being stored to a different variable before being cleared. on or otherwise `auto latest_pose = modelPoses.back();` might throw an error, in fact the original Issue was based on this, running the code without `ASSERT_FALSE` will throw an undefined behavior error, as we are clearing without checking if there are any values in `modelPoses`.
Can you help me I feel I am stuck, it's a simple issue but I am complicating things for some reason here I'm sorry about it.

Reply:
It absolutely is not the right place to check this. Line 337 clears `modelPoses`. The check in Line 354 depends on modelPoses being populated when we call `server.run`. Line 357 already checks if the `modelPoses` has data. We want to check the last element of the last run. There is no need for an assert here. Just remove these two lines.  and revert the klasst change and this PR will be good for merging.